Tests/UnitTests/Mocks/MockAttributionTypeFactory.swift

```diff
@@ -16,6 +16,25 @@ import Foundation
 #endif
 @testable import RevenueCat
 
+class MockAdClientProxy: AfficheClientProxy {
+
+    static var mockAttributionDetails: [String: NSObject] = [
+        "Version3.1":
+            [
+                "iad-campaign-id": 15292426,
+                "iad-attribution": true
+            ] as NSObject
+    ]
+    static var mockError: Error?
+    static var requestAttributionDetailsCallCount = 0
+
+    override func requestAttributionDetails(_ completionHandler: @escaping AttributionDetailsBlock) {
+        Self.requestAttributionDetailsCallCount += 1
+        completionHandler(Self.mockAttributionDetails, Self.mockError)
```

Comment:
Ditto here. Using `Result` avoids potentially setting a value and an error at the same time (or neither).